### PR TITLE
Modify Bridge config POST to accept name/value pairs

### PR DESCRIPTION
**Pull request by roncain**
Modifies the Bridge's POST handler to accept a dictionary
of name/value pairs.  Also modifies the BridgeClient to
invoke the POST using TestProperties.  This means the
Bridge is configured with the current values of the
TestProperties at runtime.

### CODE REVIEW

File: tools/Bridge/Bridge/JsonStringMediaTypeFormatter.cs [hunk 1]
```diff
@@ -77,14 +75,48 @@ public override object ReadFromStream(Type type, Stream readStream, HttpContent
...
+                sb.AppendFormat("   {0} : \"{1}\",\n", key, dictionary[key] == null ? String.Empty : dictionary[key].ToString());
+            }
+
+            sb.Remove(sb.Length - 2, 2);
+            sb.AppendLine("\n}");
+            return sb.ToString();
+        }
+
+        private static Dictionary<string, string> DeserializeDictionary(string data)
+        {
```
Comment (roncain) [inline tools/Bridge/Bridge/JsonStringMediaTypeFormatter.cs]: Reviewers: I'm not happy parsing json here but had issues trying to get the built-in WebAPI json media type formatter to work properly.  To make progress, I would like to check-in this manual parsing and later convert to normal WebAPI json deserialization (issue #214).

**Comment by roncain** [comment]
I am preparing a 2nd commit to remove use of the media type formatter.

**Comment by dmetzgar** [comment]
LGTM

File: src/System.Private.ServiceModel/tests/Common/src/BridgeClient.cs [hunk 2]
```diff
@@ -71,6 +69,24 @@ private static void MakeConfigRequest()
...
 
+        private static string CreateConfigRequestContentAsJson()
+        {
+            // Create a Json dictionary of name/value pairs from TestProperties
+            StringBuilder sb = new StringBuilder("{ ");
+            string[] propertyNames = TestProperties.PropertyNames.ToArray();
+            for (int i = 0; i < propertyNames.Length; ++i)
+            {
+                string propertyName = propertyNames[i];
+                sb.Append(String.Format("\"{0}\" : \"{1}\"", propertyName, TestProperties.GetProperty(propertyName)));
```
Comment (mconnew) [inline src/System.Private.ServiceModel/tests/Common/src/BridgeClient.cs]: sb.AppendFormat

File: tools/Bridge/Bridge/JsonSerializer.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            }
+
+            sb.Remove(sb.Length - 2, 2);
+            sb.AppendLine("\n}");
+            return sb.ToString();
+        }
+
+        internal static Dictionary<string, string> DeserializeDictionary(string data)
+        {
+            Dictionary<string, string> dictionary = new Dictionary<string, string>();
```
Comment (mconnew) [inline tools/Bridge/Bridge/JsonSerializer.cs]: DataContractJsonSerializer can do this with a dictionary, you just need to use a DataContractJsonSerializerSettings with UseSimpleDictionaryFormat set to true.

**Comment by iamjasonp** [comment]
@dotnet-bot test this please

**Comment by iamjasonp** [comment]
LGTM.

File: tools/Bridge/Bridge/JsonSerializer.cs (hunk 3 above)
Comment (roncain) [inline tools/Bridge/Bridge/JsonSerializer.cs]: I merged and then saw this.  I tested your recommendation and will consider making the change when I next touch the Bridge code.  It has some ToStrings()'s making json strings I would like to factor out into explicit serialization calls.

**Pull request merged**